ui/src/app/applications/components/application-details/application-details.tsx

```diff
@@ -723,6 +710,17 @@ export class ApplicationDetails extends React.Component<RouteComponentProps<{app
                         this.appChanged.next(app);
                     }
                 }
+            },
+            {
+                iconClassName: classNames('fa fa-redo', {'status-icon--spin': !!refreshing}),
+                title: <ActionMenuItem actionLabel='Invalidate Cache' />,
+                action: () => {
+                    if (!refreshing) {
+                        services.applications.get(app.metadata.name, app.metadata.namespace, 'hard');
+                        AppUtils.setAppRefreshing(app);
+                        this.appChanged.next(app);
+                    }
+                }
```

Comment:
Should this be disabled if !!refreshing like the Refresh Repo?

Reply:
Assuming it's legal to invalidate cache more than once, I don't want to prevent a user from trying again. The visual indicator should dissuade such action.